src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,1048 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Globalization;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    /// <summary>
+    /// Periodically polls SQL's change table to determine if any new changes have occurred to a user's table
+    /// </summary>
+    /// <remarks>
+    /// Note that there is no possiblity of SQL injection in the raw queries we generate in the Build...Command methods.
+    /// All parameters that involve inserting data from a user table are sanitized
+    /// All other parameters are generated exclusively using information about the user table's schema (such as primary key column names),
+    /// data stored in SQL's internal change table, or data stored in our own worker table.
+    /// </remarks>
+    /// <typeparam name="T">A user-defined POCO that represents a row of the user's table</typeparam>
+    internal sealed class SqlTableChangeMonitor<T> : IDisposable
+    {
+        public const string Schema = "az_func";
+        public const int BatchSize = 10;
+        public const int MaxDequeueCount = 5;
+        public const int MaxLeaseRenewalCount = 5;
+        public const int LeaseIntervalInSeconds = 30;
+        public const int PollingIntervalInSeconds = 5;
+
+        private readonly string[] variableLengthTypes = new string[] { "varchar", "nvarchar", "nchar", "char", "binary", "varbinary" };
+        private readonly string[] variablePrecisionTypes = new string[] { "numeric", "decimal" };
+
+        private readonly string _workerId;
```

Comment:
Check if the number of mutable fields in the class can be reduced, for example by passing some of them between functions, and accepting output of function via its return value rather than through shared class member variables.